### ISSUE TITLE
Cannot convert a BigInt value to a number

### ISSUE THREAD

**Issue by irzhywau** [MetaMask/eth-sig-util#298]
When I loaded the library I just got this error in a react app

```
import { recoverTypedSignature } from '@metamask/eth-sig-util';
```

I got this error when loading the page, the method is not even executed yet. any idea?

React version `18.2.0`

**Comment by coderguy1998** [comment]
I have got the same error and I guess this is being produced by the @noble package being used internally, I'm using it in an Angular project and the by keeping the buildOptimiser as true it breaks (basically the prod server) but locally everything works fine.

**Comment by mcmire** [comment]
Hi @irzhywau, when you say you receive this error in a React app, did you use `create-react-app` to create your app? Just so we can reliably reproduce this error.

**Comment by malaDev** [comment]
yes indeed I used create-react-app

**Comment by mcmire** [comment]
@malaDev Hmm, I'm not able to reproduce this in a fresh React app (`create-react-app` 5.0.1, React 18.2.0). I did get an error about Buffer not being available, but that's expected — you'll need the `buffer` package to get around this. But I didn't get the error you mentioned. Any chance you can push up a repo that reproduces this error?

**Comment by malaDev** [comment]
oh really

unfortunately it's a private repo. that's weird. do you have any idea which part can cause such an exception? any other lib?

**Comment by mcmire** [comment]
@malaDev It's possible that it's some other library. Do you have any other dependencies that you're using in your repo besides this library?

**Comment by malaDev** [comment]
I actually only get this error when I add the library, without it everything work smoothly

**Comment by mcmire** [comment]
@malaDev Right, but there may be an interaction between this library and another library you're using. If I know all of the libraries you're using then I am better able to reproduce this issue. Or, if there's a stack trace you can post, that may give me more information.
